port/unix/omrsock.c

```diff
@@ -188,6 +188,72 @@ get_omr_protocol(int32_t osProtocol)
 	return OMRSOCK_IPPROTO_DEFAULT;
 }
 
+/**
+ * @internal Map OMRSOCK API user interface socket level to
+ * OS socket level. Used to resolve the arguments of socket 
+ * option functions. 
+ * 
+ * Socket Levels currently supported are:
+ * \arg OMRSOCK_SOL_SOCKET, for most options.
+ * \arg OMRSOCK_IPPROTO_TCP, for the TCP noDelay option.
+ *
+ * @param[in] socketLevel The OMR socket level to be converted.
+ *
+ * @return OS protocol on success, or OS_SOL_SOCKET if none exists. 
+ */
+static int32_t
+get_os_socket_level(int32_t socketLevel)
+{
+	switch (socketLevel)
+	{
+		case OMRSOCK_SOL_SOCKET: 
+			return OS_SOL_SOCKET;
+		case OMRSOCK_IPPROTO_TCP: 
+			return OS_SOCK_IPPROTO_TCP;
+		default:
+			break;
+	}
```

Comment:
```suggestion
	switch (socketLevel) {
	case OMRSOCK_SOL_SOCKET: 
		return OS_SOL_SOCKET;
	case OMRSOCK_IPPROTO_TCP: 
		return OS_SOCK_IPPROTO_TCP;
	default:
		break;
	}
```

Reply:
Sounds good, I added an additional commit to modify the format for the switch statements in the rest of the file.